src/Services/IPFilterService.php

```diff
@@ -30,6 +30,8 @@ public function isBlacklisted($ip)
 
     public function blockIp($ip)
     {
+        $this->checkProvider();
+
         if ($this->dataProvider instanceof Interfaces\DatabaseDataProviderInterface) {
```

Comment:
This condition clause is not necessary because it's already checked by $this->checkProvider();
Similar to bellow functions